packages/wonder-blocks-data/src/util/__tests__/in-memory-cache.test.js

```diff
@@ -0,0 +1,396 @@
+            // Arrange
+            const cache = new InMemoryCache({
+                scope1: {
+                    key1: "a",
+                    key2: "b",
+                    key3: "a",
+                },
+                scope2: {
+                    key1: "a",
+                    key2: "b",
+                    key3: "a",
+                },
+            });
+
+            // Act
+            cache.purgeScope("scope1", (id, value) => value === "a");
+            const result = cache.clone();
+
+            // Assert
+            expect(result).toStrictEqual({
+                scope1: {
+                    key2: "b",
+                },
+                scope2: {
+                    key1: "a",
+                    key2: "b",
+                    key3: "a",
+                },
+            });
+        });
+
+        it("should remove the entire scope when there is no predicate", () => {
+            // Arrange
+            const cache = new InMemoryCache({
+                scope1: {
+                    key1: "data1",
+                    key2: "data2",
+                },
+                scope2: {
+                    key1: "data1",
+                    key2: "data2",
+                },
+            });
+
+            // Act
+            cache.purgeScope("scope1");
+            const result = cache.clone();
+
+            // Assert
+            expect(result).toStrictEqual({
+                scope2: {
+                    key1: "data1",
+                    key2: "data2",
+                },
+            });
+        });
+
+        it("should not throw if the scope does not exist", () => {
+            // Arrange
+            const cache = new InMemoryCache({
+                scope1: {
+                    key1: "data1",
+                },
+            });
+
+            // Act
+            const act = () => cache.purgeScope("scope2");
+
+            // Arrange
+            expect(act).not.toThrow();
+        });
+    });
+
+    describe("#purgeAll", () => {
+        it("should remove matching entries only", () => {
+            const cache = new InMemoryCache({
+                scope1: {key: "2"},
+                scope2: {key: "1"},
+                scope3: {key: "2"},
+            });
+
+            // Act
+            cache.purgeAll((scope, id, value) => value === "2");
+            const result = cache.clone();
+
+            // Assert
+            expect(result).toStrictEqual({
+                scope2: {key: "1"},
+            });
+        });
+
+        it("should remove the all items if there is no predicate", () => {
+            const cache = new InMemoryCache({
+                scope1: {key: "2"},
+                scope2: {key: "1"},
+                scope3: {key: "2"},
+            });
+
+            // Act
+            cache.purgeAll();
+            const result = cache.clone();
+
+            // Assert
+            expect(result).toStrictEqual({});
+        });
+    });
+
+    describe("#clone", () => {
```

Comment:
Is there a test elsewhere that checks the cloned object is structurally equivalent?

Reply:
Technically, no, but since we're using Wonder Blocks Core's `clone` method, which does test that, we're probably safe.